pkg/agent/proxy/proxier.go

```diff
@@ -455,40 +444,37 @@ func (p *proxier) installServices() {
 		}
 
 		if needUpdateEndpoints {
-			var endpointUpdateList []k8sproxy.Endpoint
-			// If the type of the Service is NodePort or LoadBalancer and both internalTrafficPolicy and externalTrafficPolicy
-			// are Local, or the type of the Service is ClusterIP and internalTrafficPolicy is Local, then only local
-			// Endpoints should be installed, otherwise all Endpoints should be installed.
-			if internalNodeLocal && (externalNodeLocal || svcInfo.NodePort() == 0) {
-				endpointUpdateList = localEndpointUpdateList
-			} else {
-				endpointUpdateList = allEndpointUpdateList
-			}
 			// Install Endpoints.
-			err := p.ofClient.InstallEndpointFlows(svcInfo.OFProtocol, endpointUpdateList)
+			err := p.ofClient.InstallEndpointFlows(svcInfo.OFProtocol, allReachableEndpoints)
 			if err != nil {
 				klog.ErrorS(err, "Error when installing Endpoints flows")
 				continue
 			}
-			if internalNodeLocal != externalNodeLocal {
-				if svcInfo.NodePort() > 0 {
+			if internalPolicyLocal != externalPolicyLocal {
+				if svcInfo.ExternallyAccessible() {
 					// If the type of the Service is NodePort or LoadBalancer, when internalTrafficPolicy and externalTrafficPolicy
-					// of the Service are different, install two groups. One group has all Endpoints, the other has only
-					// local Endpoints.
+					// of the Service are different, install two groups. One group has cluster-wide Endpoints, the
+					// other has local-wide Endpoints.
```

Comment:
"local-wide" is not a word? Probably just use "local".

Reply:
Done.